packages/react/src/components/TextField/TextField.tsx

```diff
@@ -41,83 +51,114 @@ export type TextFieldProps = {
+    const {criteria, id, type, ...rest} = props;
+
+    const [open, setOpen] = useState<boolean>(false);
+
+    const handleClick = (): void => {
       setOpen(true);
-    }
-  };
+    };
 
-  const handleClose = (): void => {
-    if (open) {
-      setOpen(false);
+    const handleClose = (): void => {
+      if (open) {
+        setOpen(false);
+      }
+    };
+
+    const tooltipContent = (): ReactNode => (
+      <List>
+        {criteria?.map((criterion: string) => (
+          <ListItem disablePadding key={criteria.indexOf(criterion)}>
+            <ListItemIcon>
+              <DoubleCircleIcon />
+            </ListItemIcon>
+            <ListItemText primary={criterion} />
+          </ListItem>
+        ))}
+      </List>
+    );
+
+    if (!criteria) {
+      return <PasswordTextField {...rest} />;
     }
-  };
-
-  const handleClickShowPassword = (): void => setShowPassword((show: boolean) => !show);
-
-  const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>): void => {
-    event.preventDefault();
-  };
-
-  const tooltipContent = (): ReactNode => (
-    <List>
-      {criteria?.map((criterion: string) => (
-        <ListItem disablePadding>
-          <ListItemIcon>
-            <DoubleCircleIcon />
-          </ListItemIcon>
-          <ListItemText primary={criterion} />
-        </ListItem>
-      ))}
-    </List>
-  );
 
-  return (
-    <div className={classes}>
-      <InputLabel htmlFor={id} aria-describedby={id} {...InputLabelProps}>
-        {label}
-      </InputLabel>
+    return (
       <Tooltip
         arrow
         placement="right-end"
         describeChild
         open={open}
         title={tooltipContent()}
         classes={{arrow: 'oxygen-text-field-tooltip-arrow', tooltip: 'oxygen-text-field-tooltip'}}
+        ref={ref}
       >
-        <MuiTextField
+        <PasswordTextField
+          ref={ref}
           id={id}
-          type={type === 'password' && (showPassword ? 'text' : 'password')}
-          InputProps={
-            type === 'password' && {
-              endAdornment: (
-                <InputAdornment position="end">
-                  <IconButton
-                    aria-label="toggle password visibility"
-                    onClick={handleClickShowPassword}
-                    onMouseDown={handleMouseDownPassword}
-                    edge="end"
-                  >
-                    {showPassword ? <VisibilityOffIcon /> : <VisibilityIcon />}
-                  </IconButton>
-                </InputAdornment>
-              ),
-            }
-          }
+          type={type}
           onClick={handleClick}
           onBlurCapture={handleClose}
           onFocus={handleClick}
           {...rest}
         />
       </Tooltip>
+    );
+  },
+) as ForwardRefExoticComponent<TextFieldProps>;
+
+const TextField: FC<TextFieldProps> & WithWrapperProps = (props: TextFieldProps): ReactElement => {
```

Comment:
This should also forward the refs.

Reply:
Fixed with https://github.com/wso2/oxygen-ui/pull/38/commits/f2ee9ffa587d3b161c78343d67540bd081438984